backend/private-graph/graph/auth.go

```diff
@@ -0,0 +1,127 @@
+package graph
```

Comment:
Introduced the auth.go file under private graph to contain auth related info.
It is cleaner, as you suggested.

I however ended up going a bit overboard with the "implementation". I tightened
up a few other things and modified the login to return user data as well as the token.